GUVNOR-2443: New Screen: Import example Project(s) from git Repository

manstis opened the pull request:
See https://issues.jboss.org/browse/GUVNOR-2443

manstis commented:
Part of an ensemble:

https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/156
https://github.com/droolsjbpm/kie-wb-common/pull/316
https://github.com/droolsjbpm/drools-wb/pull/147
https://github.com/droolsjbpm/kie-wb-distributions/pull/232

csadilek commented on kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/main/java/org/kie/workbench/common/screens/examples/backend/server/ExamplesServiceImpl.java:
```diff
@@ -0,0 +1,378 @@
+        final Set<ExampleOrganizationalUnit> exampleOrganizationalUnits = new HashSet<ExampleOrganizationalUnit>();
+        for ( OrganizationalUnit ou : organizationalUnits ) {
+            exampleOrganizationalUnits.add( new ExampleOrganizationalUnit( ou.getName() ) );
+        }
```
Assuming this is functionality for master only, you can use some new Java 8 API for things like this:
```return organizationalUnits.map(ou -> new ExampleOrganizationalUnit(ou.getName())).collect(Collectors.toSet())```...saves you 4 lines. Just a comment though and depending on preference.

csadilek commented on kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-client/src/main/java/org/kie/workbench/common/screens/examples/client/wizard/pages/organizationalunit/OUPageViewImpl.java:
```diff
@@ -0,0 +1,155 @@
+
+    public OUPageViewImpl() {
+        //Zero-argument Constructor for CDI proxies
+    }
```
This constructor shouldn't be needed as you will have an implicit no-arg constructor anyways.

manstis replied:
Sorry, yes, habitual creation of zero parameter constructors.. I'll remove.

csadilek commented on kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-client/src/main/java/org/kie/workbench/common/screens/examples/client/wizard/pages/organizationalunit/OUPageViewImpl.html:
```diff
@@ -0,0 +1,34 @@
+<div class="container-fluid">
+  <div class="row">
+    <form>
+      <div class="form-group" data-field="target-repository-form">
```
You can use id="target-repository-form" instead of data-field. data-field is one option to map to the injected fields, the others are id= and class=. the only reason to still use data-field is when you'd have multiple dom elements with the same id. In general, id is preferable.

csadilek commented on kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-client/src/main/java/org/kie/workbench/common/screens/examples/client/wizard/pages/organizationalunit/OUPage.java:
```diff
@@ -0,0 +1,195 @@
+    public OUPage() {
+        //Zero-argument constructor for CDI proxies
+    }
+
```
Not wrong, but you don't need the no-arg constructor as the other construct is safe to use with null parameters and Errai will use it instead when generating proxies.

csadilek commented on kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-client/src/main/java/org/kie/workbench/common/screens/examples/client/wizard/pages/project/ProjectPage.java:
```diff
@@ -0,0 +1,183 @@
+    public ProjectPage() {
+        //Zero-argument constructor for CDI proxies
+    }
+
```
Not wrong, but you don't need the no-arg constructor as the other construct is safe to use with null parameters and Errai will use it instead when generating proxies.

csadilek commented on kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-client/src/main/java/org/kie/workbench/common/screens/examples/client/wizard/pages/project/ProjectPageViewImpl.java:
```diff
@@ -0,0 +1,110 @@
+    public ProjectPageViewImpl() {
+        //Zero-argument Constructor for CDI proxies
+    }
+
```
constructor not needed.

csadilek commented:
@manstis all nice and clean ;). Just had a few comments inline. Happy to merge as is though. +1

manstis commented:
@csadilek I removed the zero-parameter constructors in the Views and replaced ```data-field``` with ```id``` in the ```HTML```... best to try to follow some form of best practice :)

csadilek commented:
master: 1d4821bc4e32aea2f47664135277bd8aa912f485
